tests/pthread/test_pthread_proxying_refcount.c

```diff
@@ -75,9 +84,10 @@ int main() {
   while (!executed[0] || !executed[1]) {
   }
 
-  // Break the queue abstraction to wait for the refcounts to be decreased.
-  while (*(_Atomic int*)queues[0] || *(_Atomic int*)queues[1]) {
-  }
+  // Wait 50ms for the notification to be received.
+  struct timespec ts;
+  ts.tv_nsec = 50 * 1000 * 1000;
+  nanosleep(&ts, NULL);
```

Comment:
Why 50ms?

If we have no guarantee of 50ms being enough, could we busy-wait on it forever? That seems safest. (The test runner should kill us after a minute or something like that, if we hang.)

Reply:
Hmm, maybe I don't understand the test then. It sends off 2 async tasks, and it waits for them to mark themselves as having executed. And I see we already wait for that part (`while (!executed[0] || !executed[1])`) - what else do we need to wait for? (what is "the notification" here?)

Reply:
Specifically we need to wait for the worker thread to call `em_proxying_queue_destroy` for the two queues then return to its event loop and finish receiving the two notifications corresponding to the work that had been proxied. Only once those two notifications have been fully received and the `notification` flags set to `NO_NOTIFICATION` are we ready to proceed to the next step of the test, in which the queues are freed.

Reply:
> Specifically we need to wait for the worker thread to call em_proxying_queue_destroy for the two queues then return to its event loop and finish receiving the two notifications corresponding to the work that had been proxied. Only once those two notifications have been fully received and the notification flags set to NO_NOTIFICATION are we ready to proceed to the next step of the test, in which the queues are freed.

`NO_NOTIFICATION` is set synchronously after the worker processes the queue, isn't it? Then if we set global state in the worker after it executes and destroys the queues, we know we can continue in the main thread?